File: Server/src/main/content/global/handlers/npc/DuckNPC.java
package content.global.handlers.npc;

import core.game.node.entity.npc.AbstractNPC;
import core.game.world.map.Location;
import core.plugin.Initializable;
import core.utilities.RandomFunction;

/**
 * The Duck npc.
 */
@Initializable
public final class DuckNPC extends AbstractNPC {

    private static final String FORCE_CHAT = "Quack!";

    /**
     * Instantiates a new Duck npc.
     */
    public DuckNPC() {
        super(0, null);
    }

    /**
     * Instantiates a new Duck npc.
     *
     * @param id       the id
     * @param location the location
     */
    public DuckNPC(int id, Location location) {
        super(id, location, true);
    }

    @Override
    public void tick() {
        if (RandomFunction.random(45) == 5) {
            sendChat(FORCE_CHAT);
        }
        super.tick();
    }

    @Override
    public AbstractNPC construct(int id, Location location, Object... objects) {
        return new DuckNPC(id, location);
    }

    @Override
    public int[] getIds() {
        return new int[]{46, 2693, 6113};
    }

}

File: Server/src/main/content/global/handlers/npc/DuckNPC.kt
package content.global.handlers.npc

import core.api.consts.NPCs
import core.api.sendChat
import core.game.node.entity.npc.NPC
import core.game.node.entity.npc.NPCBehavior
import core.utilities.RandomFunction

class DuckNPC : NPCBehavior(NPCs.DUCK_46, NPCs.DUCK_2693, NPCs.DUCK_6113) {

    override fun tick(self: NPC): Boolean {
        if (RandomFunction.roll(100)) {
            sendChat(self, when(self.id) {
                46 -> "Eep!"
                else -> "Quack!"
            })
        }
        return true
    }
}
